cmd/neofs-adm/internal/modules/storagecfg/root.go

```diff
@@ -357,13 +350,21 @@ func depositGas(cmd *cobra.Command, acc *wallet.Account, network string) {
 	neofsHash, _ := util.Uint160DecodeStringLE(n3config[network].NeoFSContract)
 
 	gasHash, err := mainClient.GetNativeContractHash(nativenames.Gas)
-	fatalOnErr(err)
+	if err != nil {
+		fatalOnErr(fmt.Errorf("gas contract hash: %w", err))
+	}
 
-	tx, err := mainClient.CreateNEP17TransferTx(acc, neofsHash, gasHash, amount.Int64(), 0, nil, nil)
-	fatalOnErr(err)
+	mainActor, err := actor.NewSimple(mainClient, acc)
+	if err != nil {
+		fatalOnErr(fmt.Errorf("creating actor over main chain client: %w", err))
+	}
 
-	txHash, err := mainClient.SignAndPushTx(tx, acc, nil)
-	fatalOnErr(err)
+	mainGas := nep17.New(mainActor, gasHash)
+
+	txHash, _, err := mainGas.Transfer(accSH, neofsHash, amount, nil)
```

Comment:
BTW, you can use the second returned value (`ValidUntilBlock`) for precise transaction acceptance tracking (instead of 20-seconds timer). If `VUB`-th block is accepted but transaction's application log is not yet available, then the transaction won't be accepted at all.

Reply:
BTW, isn't it 30 seconds? Maybe it's related to nspcc-dev/neo-go#2663?

Reply:
Yeah, it's very risky, any minimal delay in block acceptance can cause a timeout this way.

Reply:
https://github.com/nspcc-dev/neofs-node/issues/1741